feat(ui): release notes modal

peps opened the pull request:
This adds a Release Notes modal that pops up after a new app version is detected.

![CleanShot 2025-01-20 at 17 22 50@2x](https://github.com/user-attachments/assets/b88e6569-6a2f-4cb1-ab65-b79ad82871e4)

Functionality requirements:

- The modal will trigger after the Setup screen on the main dashboard
- Only show the popup if the app version and fetched release notes version are a match.
- Only show the popup once per version.
- Only fetch the release notes on startup if the versions are not a match

PanchoBubble commented on src/containers/floating/Settings/sections/releaseNotes/useReleaseNotes.ts:
```diff
@@ -0,0 +1,56 @@
+        if (!triggerEffect) return;
+
+        const currentAppVersion = packageInfo.version;
+
+        // TODO: Need to save the fetched releaseNotesVersion to persistant storage
+        //       and compare it with the appVersion to determine if the release notes
+        //       dialog has already been shown to the user.
```
```suggestion
```

peps replied:
done

shanimal08 commented on src/containers/phase/Setup/Setup.tsx:
```diff
@@ -5,9 +5,12 @@ import InfoNav from './components/InfoNav/InfoNav';
 
 export default function Setup() {
     useSetUp();
+    useReleaseNotes({ triggerEffect: true });
```
LGTM & tested locally with the admin trigger

just a bit concerned about this one in during setup and how it will react with auto update stuff? 😅  might be good for @mmrrnn to have a look through too

peps replied:
@shanimal08 good point, I can move the trigger for the modal to happen after setup completes. That way it wont conflict with the updater.

mmrrnn replied:
Why not to invoke `check_for_updates` to ensure there is no update available?

mmrrnn replied:
We can also add it to the `get_last_changelog_version` response

peps replied:
@mmrrnn I moved it to the `MainView` so it shouldn't conflict with the Updater anymore. 

One weird requirement for this is that `CHANGELOG.md` might not be updated in time for a release. So this will pop only when the `latestVersion` in `CHANGELOG.md` matches the `appVersion` number, and when `latestVersion` does not equal `get_last_changelog_version`

mmrrnn replied:
Yeah, that's fine :+1:

peps commented:
@brianp @shanimal08 I've updated the modal to trigger in `MainView` instead of `Setup`, so it should no longer conflict with the update modal. 

I also updated the `Markdown` styles for the latest release notes

peps commented:
Note: The layout had been changed a bit. 

- Simplified Header
- Got It confirmation button that closes the modal on click
- Scrollbar added

brianp commented:
Mentioned in DM, but just to add here we're going to revisit this PR and bring all the logic from the front end to the backend. @Misieq01 will look to address this before we merge it.

brianp commented:
Closing in favour of #1454
